src/affordance/tasks/tools.py

```diff
@@ -25,7 +25,49 @@
 - (str) the output
 - (str) runtime details to be logged
 """
+import pickle
 
+def get_run_dict(run_title):
+    try:
+        with open(f'{run_title}.pickle', 'rb') as handle:
+            run_dict = pickle.load(handle)
+    except FileNotFoundError:
+        run_dict = {}
+        print('run dict not found, creating a new one!')
+    return run_dict
+
+def save_run_dict(run_dict, run_title):
+    with open(f'{run_title}.pickle', 'wb') as handle:
+        pickle.dump(run_dict, handle, protocol=pickle.HIGHEST_PROTOCOL)
+    return
+
+
+def read(key, run_title) -> tuple(str, str):
+    read_write_dict = get_run_dict(run_title)
+    if key in read_write_dict:
+        return read_write_dict[key], None
+    #TODO finalize key not found behaviour
+    else:
+        return ', '.join(read_write_dict.keys()), None
+    return f"<key \'{key}\' did not exist>", None  # TODO: return a useful log message
```

Comment:
These two returns are redundant, and whichever one you keep doesn't need to be in an else block